tool/src/org/antlr/v4/tool/GrammarParserInterpreter.java

```diff
@@ -64,7 +67,7 @@ public GrammarParserInterpreter(Grammar g, ATN atn, TokenStream input) {
 			  input);
 		this.g = g;
 		decisionStatesThatSetOuterAltNumInContext = findOuterMostDecisionStates();
-		stateToAltsMap = new int[g.atn.getNumberOfDecisions()][];
+		stateToAltsMap = new int[g.atn.states.size()][];
```

Comment:
:thought_balloon: You could use `DecisionState.decision` as the key instead of `ATNState.stateNumber` if you want to keep this array the smaller size.